server/chat-commands/admin.ts

```diff
@@ -1154,17 +1154,43 @@ export const commands: Chat.ChatCommands = {
 		`/endemergency - Turns off emergency mode. Requires: &`,
 	],
 
-	kill(target, room, user) {
+	async savebattles(target, room, user) {
+		this.checkCan('rangeban'); // admins can restart, so they should be able to do this if needed
+		this.sendReply(`Saving battles...`);
+		const count = await Rooms.global.saveBattles();
+		this.sendReply(`DONE.`);
+		this.sendReply(`${count} battles saved.`);
+		this.addModAction(`${user.name} used /savebattles`);
+	},
+
+	async kill(target, room, user) {
 		this.checkCan('lockdown');
+		let noSave = toID(target) === 'nosave';
+		if (!Config.usepostgres) noSave = true;
 
-		if (Rooms.global.lockdown !== true) {
-			return this.errorReply("For safety reasons, /kill can only be used during lockdown.");
+		if (Rooms.global.lockdown !== true && noSave) {
+			return this.errorReply("For safety reasons, using /kill without saving battles can only be done during lockdown.");
 		}
 
 		if (Monitor.updateServerLock) {
 			return this.errorReply("Wait for /updateserver to finish before using /kill.");
 		}
 
+		if (!noSave) {
+			this.sendReply('Saving battles...');
+			Rooms.global.lockdown = true; // we don't want more battles starting while we save
+			for (const u of Users.users.values()) {
+				u.send(
+					`|pm|&|${u.getIdentity()}|/raw <div class="broadcast-red"><b>The server is restarting soon.</b><br />` +
+					`While battles are being saved, no more can be started. <br />` +
+					`After the restart, you will be able to resume your battles from where you left off.`
```

Comment:
This should probably mention something about how your battles may be frozen after saving but before the restart? Otherwise, I think the message about your battle being paused is going to confuse people.

Reply:
Done.